system-tests/lib/docker.ts

```diff
@@ -57,7 +57,9 @@ class DockerProcess extends EventEmitter implements SpawnerResult {
 
     for (const k in opts.env) {
       // skip problematic env vars that we don't wanna preserve from `process.env`
-      if (['DISPLAY', 'USER', 'HOME', 'USERNAME', 'PATH'].includes(k)) continue
+      if (
+        ['DISPLAY', 'USER', 'HOME', 'USERNAME', 'PATH'].includes(k)
+        || k.startsWith('npm_')) continue
```

Comment:
Very small nit, but now this is more than one line:

```suggestion
      if (
        ['DISPLAY', 'USER', 'HOME', 'USERNAME', 'PATH'].includes(k)
        || k.startsWith('npm_')
      ) {
        continue
      }
```